addons/stock/models/stock_scrap.py

```diff
@@ -139,8 +139,8 @@ def do_scrap(self):
             scrap.name = self.env['ir.sequence'].next_by_code('stock.scrap') or _('New')
             move = self.env['stock.move'].create(scrap._prepare_move_values())
             # master: replace context by cancel_backorder
-            move.with_context(is_scrap=True)._action_done()
-            scrap.write({'move_id': move.id, 'state': 'done'})
+            moves = move.with_context(is_scrap=True)._action_done()
+            scrap.write({'move_ids': [move.id for move in moves], 'state': 'done'})
```

Comment:
you could set scrap_id in _prepare_move_values. Instead of writing the one2many field it will prevent to trigger 2 write instead if 1